x-pack/plugins/security_solution/public/resolver/store/data/visible_entities.test.ts

```diff
@@ -0,0 +1,159 @@
+
+describe('resolver visible entities', () => {
+  let processA: LegacyEndpointEvent;
+  let processB: LegacyEndpointEvent;
+  let processC: LegacyEndpointEvent;
+  let processD: LegacyEndpointEvent;
+  let processE: LegacyEndpointEvent;
+  let processF: LegacyEndpointEvent;
+  let processG: LegacyEndpointEvent;
+  let store: Store<ResolverState, ResolverAction>;
+
+  beforeEach(() => {
+    /*
+     *          A
+     *          |
+     *          B
+     *          |
+     *          C
+     *          |
+     *          D etc
+     */
+    processA = mockProcessEvent({
+      endgame: {
+        process_name: '',
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 0,
+      },
+    });
+    processB = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'already_running',
+        unique_pid: 1,
+        unique_ppid: 0,
+      },
+    });
+    processC = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 2,
+        unique_ppid: 1,
+      },
+    });
+    processD = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 3,
+        unique_ppid: 2,
+      },
+    });
+    processE = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 4,
+        unique_ppid: 3,
+      },
+    });
+    processF = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 5,
+        unique_ppid: 4,
+      },
+    });
+    processF = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 6,
+        unique_ppid: 5,
+      },
+    });
+    processG = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'termination_event',
+        unique_pid: 7,
+        unique_ppid: 6,
+      },
+    });
+    store = createStore(resolverReducer, undefined);
+  });
+  describe('when rendering a large tree with a small viewport', () => {
+    beforeEach(() => {
+      const events: ResolverEvent[] = [processA, processB];
+      const action: ResolverAction = {
+        type: 'serverReturnedResolverData',
+        events,
+        stats: new Map(),
+      };
+      const cameraAction: ResolverAction = { type: 'userSetRasterSize', payload: [300, 200] };
+      store.dispatch(action);
+      store.dispatch(cameraAction);
+    });
+    it('the visibleProcessNodePositions list should only include 2 nodes', () => {
+      const { processNodePositions } = visibleProcessNodePositionsAndEdgeLineSegments(
+        store.getState()
+      )(Date.now());
```

Comment:
If you don't an animation running, you can just use `0`. that eliminates the code smell of side effects